libraries/lib-stretching-sequence/tests/TestWaveClipMaker.h

```diff
@@ -0,0 +1,45 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  TestWaveClipMaker.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#pragma once
+
+#include "AudioContainer.h"
+#include "SampleFormat.h"
+#include "WaveClip.h"
+#include "WaveTrack.h"
```

Comment:
`lib-stretching-sequence` does not depend on `lib-wave-track`, but the opposite.

However the test executable depends on `lib-wave-track`.

The build system let you do this, but I think I don't like it and should consider it a cycle.

Instead, you could lift all testing files that depend on WaveTrack and WaveClip into a new `lib-wave-track/test` directory.

That may mean `lib-wave-track/test` might need to used some of the mocks defined in `lib-stretching-sequence/test`.

Does the test build system allow that dependency?  @crsib ?

Reply:
I don't think we have a cyclic dependency. In fact, if I could remove `lib-stretching-sequence` from the list of libraries needed by this test executable and it would still compile. It's just one unnecessary arrow in the dependency graph ;)

Reply:
But I think the test directory for a library should be responsible to test only that library.

If a test executable involves other libraries higher in the graph, that argues for moving those tests.